### ISSUE TITLE
The type initializer for 'Microsoft.Data.SqlClient.SNINativeMethodWrapper' threw an exception. after upgrade from EFCore 2.2 to 3.1

### ISSUE THREAD

**Issue by ssteiner** [dotnet/efcore#19172]
Now that EFCore 3.1 supports .NET standard 2.0 I wanted to upgrade my app that has been using EFCore 2.2 so far. so I swapped out the 4 packages I'm using (Microsoft.EntityFrameworkCore.Design/Proxies/SqlServer/Tools - all version 3.1.0 now) and gave it a go.

At startup, I check if the connection to the database is valid using this approach

```C#
	using (AudmDatabaseContext db = new AudmDatabaseContext(connectionString))
	{
		try
		{
			await db.Database.OpenConnectionAsync(token).ConfigureAwait(false);
			return new AudmOperationResult { IsSuccess = true };
		}
		catch (Exception e)
		{
			Log($"Exception trying to connect to database: {e.Message}", 2, userInfo);
		}
	}
```

where AudmDatabaseContext derives from DbContext

```C#
    public partial class AudmDatabaseContext : DbContext
    {
        private string connectionString;

        public AudmDatabaseContext(string connectionString = null)
            : base()
        {
            this.connectionString = connectionString;
        }

        protected override void OnConfiguring(DbContextOptionsBuilder optionsBuilder)
        {
            if (string.IsNullOrEmpty(connectionString))
            {
                string connectionString = ConfigurationManager.ConnectionStrings["AudmDatabase"].ConnectionString;
                optionsBuilder.UseSqlServer(connectionString);
            }
            else
                optionsBuilder.UseSqlServer(connectionString);
            optionsBuilder.UseLazyLoadingProxies();
            optionsBuilder.EnableSensitiveDataLogging();
            base.OnConfiguring(optionsBuilder);
        }
    }
```

And connectionstring being

```
Data Source=(local);Initial Catalog=AUDM_Microsoft2;Integrated Security=True;MultipleActiveResultSets=True;Application Name=EntityFramework
```

the Inner exception tells me this

Failed to load D:\Personal\Documents\Visual Studio 2012\Projects\UnifiedProvisioning\UnifiedUCProvisioning\bin\Debug MS\x64\SNI.dll

**To reproduce**
Working on a repro project right now.

**Expected behavior**
Code works as prior to the upgrade.

**Environment**
.NET Target: .NET Standard 2.0 class library, consumed by a .NET 4.7.2 application
OS: Win10 1909

**Comment by ajcvickers** [comment]
@ssteiner I don't see anything obvious here; I think we will need the repro project.

**Comment by ajcvickers** [comment]
**EF Team Triage:** Closing this issue as the requested additional details have not been provided and we have been unable to reproduce it.

*BTW this is a canned response and may have info or details that do not directly apply to this particular issue. While we'd like to spend the time to uniquely address every incoming issue, we get a lot traffic on the EF projects and that is not practical. To ensure we maximize the time we have to work on fixing bugs, implementing new features, etc. we use canned responses for common triage decisions.*

**Comment by ssteiner** [comment]
Sorry for the late reply - shortly after my report I went on an extended holiday. 
I attempted to create a repro project, but the repro project works fine. Even though I have a standalone DBL project, it is still rather massive. So instead of attempting to strip that down, any ideas what could trigger such a thing so I have something to go on to try and attempt to build a repro project that actually reproduces the issue?
